url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_SelectClauseParser.java

```diff
@@ -0,0 +1,7251 @@
+			stream_KW_INCRE.add(KW_INCRE73);
+
+			LPAREN74=(Token)match(input,LPAREN,FOLLOW_LPAREN_in_incrementalClause1054);  
+			stream_LPAREN.add(LPAREN74);
+
+			pushFollow(FOLLOW_incrementalArgs_in_incrementalClause1056);
+			incrementalArgs75=incrementalArgs();
+			state._fsp--;
+
+			stream_incrementalArgs.add(incrementalArgs75.getTree());
+			RPAREN76=(Token)match(input,RPAREN,FOLLOW_RPAREN_in_incrementalClause1058);  
+			stream_RPAREN.add(RPAREN76);
+
+			STAR77=(Token)match(input,STAR,FOLLOW_STAR_in_incrementalClause1060);  
+			stream_STAR.add(STAR77);
+
+			DIVIDE78=(Token)match(input,DIVIDE,FOLLOW_DIVIDE_in_incrementalClause1062);  
+			stream_DIVIDE.add(DIVIDE78);
+
+			// AST REWRITE
+			// elements: incrementalArgs
+			// token labels: 
+			// rule labels: retval
+			// token list labels: 
+			// rule list labels: 
+			// wildcard labels: 
+			retval.tree = root_0;
+			RewriteRuleSubtreeStream stream_retval=new RewriteRuleSubtreeStream(adaptor,"rule retval",retval!=null?retval.getTree():null);
+
+			root_0 = (CommonTree)adaptor.nil();
+			// 173:6: -> ^( TOK_INCRE ( incrementalArgs )? )
+			{
+				// SelectClauseParser.g:173:10: ^( TOK_INCRE ( incrementalArgs )? )
+				{
+				CommonTree root_1 = (CommonTree)adaptor.nil();
+				root_1 = (CommonTree)adaptor.becomeRoot((CommonTree)adaptor.create(TOK_INCRE, "TOK_INCRE"), root_1);
+				// SelectClauseParser.g:173:22: ( incrementalArgs )?
+				if ( stream_incrementalArgs.hasNext() ) {
+					adaptor.addChild(root_1, stream_incrementalArgs.nextTree());
+				}
+				stream_incrementalArgs.reset();
+
+				adaptor.addChild(root_0, root_1);
+				}
+
+			}
+
+
+			retval.tree = root_0;
+
+			}
+
+			retval.stop = input.LT(-1);
+
+			retval.tree = (CommonTree)adaptor.rulePostProcessing(root_0);
+			adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+
+			 gParent.msgs.pop(); 
+		}
+
+		catch (RecognitionException e) {
+		  throw e;
+		}
+
+		finally {
+			// do for sure before leaving
+		}
+		return retval;
+	}
+	// $ANTLR end "incrementalClause"
+
+
+	public static class incrementalArgs_return extends ParserRuleReturnScope {
+		CommonTree tree;
+		@Override
+		public CommonTree getTree() { return tree; }
+	};
+
+
+	// $ANTLR start "incrementalArgs"
+	// SelectClauseParser.g:176:1: incrementalArgs : (bdate= betweenDate -> ^( TOK_DATE $bdate) | KW_CONSTANT time= TimeUnit (interval= KW_INTERVAL numerator= Number DIVIDE unit1= ( KW_DAY | KW_HOUR | KW_MINUTE | KW_SECOND ) )? -> {interval != null}? ^( TOK_CONSTANT $time TOK_INTERVAL $numerator $unit1) -> ^( TOK_CONSTANT $time) | KW_AFTER sdate= startDate (interval= KW_INTERVAL numerator= Number DIVIDE unit2= ( KW_DAY | KW_HOUR | KW_MINUTE | KW_SECOND ) )? -> {interval != null}? ^( TOK_DATE $sdate TOK_INTERVAL $numerator $unit2) -> ^( TOK_DATE $sdate) );
+	public final HiveParser_SelectClauseParser.incrementalArgs_return incrementalArgs() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 199 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.